packages/auth/src/legacy-token.ts

```diff
@@ -0,0 +1,64 @@
+import {
+  createCipheriv,
+  createDecipheriv,
+  HexBase64BinaryEncoding,
+  randomBytes,
+  Utf8AsciiBinaryEncoding,
+} from 'crypto';
+import { TOKEN_VALID_LENGTH } from '@verdaccio/config';
+import buildDebug from 'debug';
+
+const debug = buildDebug('verdaccio:auth:token:legacy');
+
+export const defaultAlgorithm = process.env.VERDACCIO_LEGACY_ALGORITHM || 'aes-256-ctr';
+const inputEncoding: Utf8AsciiBinaryEncoding = 'utf8';
+const outputEncoding: HexBase64BinaryEncoding = 'hex';
+// For AES, this is always 16
+const IV_LENGTH = 16;
+// Must be 256 bits (32 characters)
+// https://stackoverflow.com/questions/50963160/invalid-key-length-in-crypto-createcipheriv#50963356
+const VERDACCIO_LEGACY_ENCRYPTION_KEY = process.env.VERDACCIO_LEGACY_ENCRYPTION_KEY;
+
+export function aesEncrypt(value: string, key: string): string | void {
+  // https://nodejs.org/api/crypto.html#crypto_crypto_createcipher_algorithm_password_options
+  // https://www.grainger.xyz/changing-from-cipher-to-cipheriv/
+  debug('encrypt %o', value);
+  debug('algorithm %o', defaultAlgorithm);
+  const iv = Buffer.from(randomBytes(IV_LENGTH));
+  const secretKey = VERDACCIO_LEGACY_ENCRYPTION_KEY || key;
+  const isKeyValid = key?.length === TOKEN_VALID_LENGTH;
```

Comment:
Should this check `secretKey` or just `key`?

Reply:
yeah, is the same. I can also use `secretKey` if you think is more readable.